panacea/test/panacea/pml/analysis/ddis_test.exs

```diff
@@ -0,0 +1,218 @@
+defmodule Panacea.Pml.Analysis.DdisTest do
+  alias Panacea.Pml.Analysis
+  use ExUnit.Case
+
+  defp parse_pml(pml) do
+    {:ok, ast} = Panacea.Pml.Parser.parse(pml)
+    ast
+  end
+
+  def test_ddis() do
+    [
+      %{"drug_a" => "http://purl.com/paracetamol", "drug_b" => "http://purl.com/cocaine"}
+    ]
+  end
+
+  def test_drugs() do
+    [
+      %{"label" => "paracetamol", "uri" => "http://purl.com/paracetamol"},
+      %{"label" => "cocaine", "uri" => "http://purl.com/cocaine"},
+    ]
+  end
+
+  describe "run/2" do
+    test "it categorizes sequential DDIs correctly" do
+      pml = """
+      process sequential_ddis {
+        action s1 {
+          requires { drug { "paracetamol" } }
+        }
+        action s2 {
+          requires { drug { "cocaine" } }
+        }
+      }
+      """
+      ast = parse_pml(pml)
+
+      assert Analysis.Ddis.run(ast, test_ddis(), test_drugs()) == [
+        %{
+          "category" => :sequential,
+          "drug_a" => "http://purl.com/paracetamol",
+          "drug_b" => "http://purl.com/cocaine",
+          "enclosing_construct" => %{
+            "type" => :process,
+            "line" => 1
+          }
+        }
+      ]
+    end
+
+    test "it categorizes parallel DDIs correctly" do
+      pml = """
+      process parallel_ddis {
+        branch {
+          action b1 {
+            requires { drug { "paracetamol" } }
+          }
+          action b2 {
+            requires { drug { "cocaine" } }
+          }
+        }
+      }
+      """
+      ast = parse_pml(pml)
+
+      assert Analysis.Ddis.run(ast, test_ddis(), test_drugs()) == [
+        %{
+          "category" => :parallel,
+          "drug_a" => "http://purl.com/paracetamol",
+          "drug_b" => "http://purl.com/cocaine",
+          "enclosing_construct" => %{
+            "type" => :branch,
+            "line" => 2
+          }
+        }
+      ]
+    end
+
+    test "it categorizes alternative DDIs correctly" do
+      pml = """
+      process alternative_ddis {
+        selection {
```

Comment:
Yeah - it should be noted that `alternative` here means `Alternative non-DDI` from the backlog.

As you said, `Alternative DDIs` would have to be handled separately to all of the other DDI categories.

The current approach will label any interacting drug whose closets common ancestor is a selection as `alternative`. The customer didn't buy either of the `Alternative X` features, so I went with this simplified approach.